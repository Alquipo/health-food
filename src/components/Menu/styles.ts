import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.nav`
  height: 10rem;
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  position: relative;

  a {
    text-decoration: none;
    width: 100%;
  }
`

export const Logo = styled.h1`
  ${({ theme }) => css`
    color: ${theme.colors.primary};
    font-size: ${theme.spacings.medium};
  `}
`

type MenuNavProps = {
  isOpen: boolean
}

const menuNavModifiers = {
  isOpen: () => css`
    animation: menuAnimation 0.3s forwards;

    @keyframes menuAnimation {
      to {
        opacity: 1;
        pointer-events: all;
      }
    }
  `
}

export const MenuNav = styled.div<MenuNavProps>`
  ${({ theme, isOpen }) => css`
    color: ${theme.colors.white};

    ${media.lessThan('medium')`
    position: absolute;
      opacity: 0;
      pointer-events: none;
      top: 100%;
      right: 0;
      background: ${theme.colors.primary};
      z-index: ${theme.layers.menu};

      ${isOpen && menuNavModifiers.isOpen()}
    `};
  `}
`

export const MenuLink = styled.a`
  ${({ theme }) => css`
    margin-right: ${theme.spacings.small};
    color: ${theme.colors.white};
    text-decoration: none;
    text-transform: uppercase;
    font-size: ${theme.font.sizes.small};
    transition: ${theme.transition.fast};

    &:hover {
      filter: brightness(0.95);
    }

    ${media.lessThan('large')`
    color: ${theme.colors.text};

    `};

    ${media.lessThan('medium')`
      color: ${theme.colors.white};
      display: block;
      padding: ${theme.spacings.xsmall} ${theme.spacings.small};
      border-bottom: solid 0.1rem ${theme.colors.white};
      width: 100%;
    `};
  `}
`

export const Register = styled.button`
  ${({ theme }) => css`
    border: none;
    text-transform: uppercase;
    font-size: ${theme.font.sizes.small};
    transition: ${theme.transition.fast};
    padding: ${theme.spacings.xxsmall} ${theme.spacings.xsmall};
    background: ${theme.colors.white};
    border-radius: ${theme.border.radius};
    color: ${theme.colors.primary};
    cursor: pointer;
    font-weight: ${theme.font.bold};

    ${media.lessThan('large')`
      background: ${theme.colors.primary};
      color: ${theme.colors.white};

    `};

    ${media.greaterThan('medium')`
      padding: ${theme.spacings.xsmall} ${theme.spacings.small};
    `};
  `}
`

export const MobileMenuButton = styled.button`
  ${({ theme }) => css`
    background: transparent;
    border: none;
    cursor: pointer;

    svg {
      fill: ${theme.colors.secondary};
    }

    ${media.greaterThan('medium')`
      display: none;
    `};
  `}
`
