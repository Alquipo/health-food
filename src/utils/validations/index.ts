import * as yup from 'yup'

const nascimentoRegex =
  /^((((19|[2-9]\d)\d{2})-(0[13578]|1[02])-(0[1-9]|[12]\d|3[01]))|(((19|[2-9]\d)\d{2})-(0[13456789]|1[012])-(0[1-9]|[12]\d|30))|(((19|[2-9]\d)\d{2})-02-(0[1-9]|1\d|2[0-8]))|(((1[6-9]|[2-9]\d)(0[48]|[2468][048]|[13579][26])|((16|[2468][048]|[3579][26])00))-02-29))$/g

const cpfRegex = /^\d{3}.?\d{3}.?\d{3}-?\d{2}$/

const cepRegex = /^([\d]{2})\.*([\d]{3})-*([\d]{3})/

const estadoRegex =
  /^(AC|AL|AP|AM|BA|CE|DF|ES|GO|MA|MT|MS|MG|PA|PB|PR|PE|PI|RJ|RN|RS|RO|RR|SC|SP|SE|TO)$/i

export const fieldsValidationSchema = yup.object({
  name: yup.string().required('Is not allowed to be empty'),
  birthDate: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(nascimentoRegex, 'must be a valid email'),
  cpf: yup
    .string()
    .max(14, 'Must be 14 characters')
    .required('Is not allowed to be empty')
    .matches(cpfRegex, 'Invalid CPF'),
  zipCode: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(cepRegex, 'Invalid ZipCode'),
  address: yup.string().required('Is not allowed to be empty'),
  district: yup.string().required('Is not allowed to be empty'),
  number: yup.number().required('Is not allowed to be empty'),
  city: yup.string().required('Is not allowed to be empty'),
  state: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(estadoRegex, 'Invalid State')
})
