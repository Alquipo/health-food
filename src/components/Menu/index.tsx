import { useState } from 'react'
import Link from 'next/link'

import * as S from './styles'

import { Menu as MobileMenuIcon } from '@styled-icons/boxicons-regular'

export default function Menu() {
  const [isOpen, setIsOpen] = useState(false)

  /* istanbul ignore next */
  function handleMobileMenu() {
    setIsOpen((state) => !state)
  }

  return (
    <S.Wrapper>
      <S.Logo>Healthy Food</S.Logo>

      <S.MenuNav isOpen={isOpen} aria-hidden={isOpen}>
        <S.MenuLink href="#">Healthy Recipes</S.MenuLink>
        <S.MenuLink href="#">Blog</S.MenuLink>
        <S.MenuLink href="#">Join</S.MenuLink>

        <Link href="/sign-up">
          <a>
            <S.Register>Register</S.Register>
          </a>
        </Link>
      </S.MenuNav>

      <S.MobileMenuButton onClick={handleMobileMenu}>
        <MobileMenuIcon size={40} aria-label="Open Menu" />
      </S.MobileMenuButton>
    </S.Wrapper>
  )
}
