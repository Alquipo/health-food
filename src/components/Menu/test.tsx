import { fireEvent, render, screen, waitFor } from 'utils/test-utils'

import Menu from '.'

// eslint-disable-next-line @typescript-eslint/no-var-requires
const useRouter = jest.spyOn(require('next/router'), 'useRouter')

useRouter.mockImplementation(() => ({
  query: {}
}))

describe('<Menu/>', () => {
  it('should render a Menu correctly', () => {
    const { container } = render(<Menu />)

    expect(screen.getByLabelText(/open menu/i)).toBeInTheDocument()

    expect(
      screen.getByRole('heading', {
        name: /Healthy Food/i
      })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('link', {
        name: /Healthy Recipes/i
      })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('link', {
        name: /Blog/i
      })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('link', {
        name: /Join/i
      })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('link', {
        name: /Register/i
      })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should handle submit button ', () => {
    render(<Menu />)

    const handleMobileMenu = jest.fn()

    fireEvent.click(screen.getByLabelText(/Open Menu/i))

    waitFor(() => {
      expect(handleMobileMenu).toHaveBeenCalled()
    })
  })
})
