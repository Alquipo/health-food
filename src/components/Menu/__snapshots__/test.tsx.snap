// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu/> should render a Menu correctly 1`] = `
.c0 {
  height: 10rem;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  margin: 0 auto;
  max-width: 120rem;
  padding: 0 1.5rem;
}

.c0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
}

.c1 {
  color: #BADC58;
  font-size: 3.2rem;
}

.c2 {
  color: #FFFFFF;
}

.c3 {
  margin-right: 2.4rem;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  font-size: 1.8rem;
  font-weight: 600;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
}

.c3:hover {
  -webkit-filter: brightness(0.95);
  filter: brightness(0.95);
}

.c4 {
  border: none;
  text-transform: uppercase;
  font-size: 1.6rem;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
  padding: 0.8rem 1.6rem;
  background: #FFFFFF;
  border-radius: 0.6rem;
  color: #BADC58;
  cursor: pointer;
  font-weight: 600;
}

.c5 {
  background: transparent;
  border: none;
  cursor: pointer;
}

.c5 svg {
  fill: #1D164D;
}

.c6 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

@media (max-width:768px) {
  .c2 {
    position: absolute;
    opacity: 0;
    pointer-events: none;
    top: 100%;
    right: 0;
    background: #BADC58;
    z-index: 20;
  }
}

@media (max-width:1170px) {
  .c3 {
    color: #9E9BAF;
  }
}

@media (max-width:768px) {
  .c3 {
    color: #FFFFFF;
    display: block;
    padding: 1.6rem 2.4rem;
    border-bottom: solid 0.1rem #FFFFFF;
    width: 100%;
  }
}

@media (max-width:1170px) {
  .c4 {
    background: #BADC58;
    color: #FFFFFF;
  }
}

@media (min-width:768px) {
  .c4 {
    padding: 1.6rem 2.4rem;
  }
}

@media (min-width:768px) {
  .c5 {
    display: none;
  }
}

<menu
  class="c0"
>
  <h1
    class="c1"
  >
    Healthy Food
  </h1>
  <nav
    aria-hidden="false"
    class="c2"
  >
    <a
      class="c3"
      href="#"
    >
      Healthy Recipes
    </a>
    <a
      class="c3"
      href="#"
    >
      Blog
    </a>
    <a
      class="c3"
      href="#"
    >
      Join
    </a>
    <a
      href="/sign-up"
    >
      <button
        class="c4"
      >
        Register
      </button>
    </a>
  </nav>
  <button
    class="c5"
  >
    <svg
      aria-hidden="true"
      aria-label="Open Menu"
      class="c6"
      fill="currentColor"
      focusable="false"
      height="40"
      viewBox="0 0 24 24"
      width="40"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4 6h16v2H4zm0 5h16v2H4zm0 5h16v2H4z"
      />
    </svg>
  </button>
</menu>
`;
